store/sqlstore/post_store.go

```diff
@@ -2411,12 +2464,10 @@ func (s *SqlPostStore) cleanupThreads(postId, rootId string, permanent bool, use
 			if s.DriverName() == model.DatabaseDriverPostgres {
 				updateQuery = updateQuery.Set("Participants", sq.Expr("Participants - ?", userId))
 			} else {
-				// The .Where is because JSON_REMOVE returns null if the element to remove wasn't present
 				updateQuery = updateQuery.
 					Set("Participants", sq.Expr(
-						`JSON_REMOVE(Participants, JSON_UNQUOTE(JSON_SEARCH(Participants, 'one', ?)))`, userId,
-					)).
-					Where(sq.Expr(`JSON_CONTAINS(Participants, ?)`, strconv.Quote(userId)))
```

Comment:
I had to remove the `WHERE` clause to support calling the cleanup from a loop in the same transaction (the case of `deleteAllCommentsByUser`.

Case:

- User has made 2 comments in a thread.
- We permanent delete all posts by User.
- `deleteAllCommentByUser` deletes at first all comments and then
- `cleanupThreads` is called in a loop for each of those 2 comments.
- The first time it will remove User from participants and set ReplyCount -1.
- The second time the Where clause would have failed and the Thread would not update
- resulting in a ReplyCount that's not correct.

By just removing the Where clause in mysql Participants were set to `null` which resulted in a string to null conversion error.
Adding `IFNULL` fixes that.

Reply:
Excellent catch, I missed this in the original code review.